subsystems/TypeHandlerLibrary/src/main/java/org/terasology/persistence/typeHandling/coreTypes/GenericMapTypeHandler.java

```diff
@@ -61,26 +61,45 @@ private PersistedData serializeEntry(Map.Entry<K, V> entry, PersistedDataSeriali
 
     @Override
     public Optional<Map<K, V>> deserialize(PersistedData data) {
-        if (!data.isArray()) {
+        if (!data.isArray() || data.isValueMap()) {
```

Comment:
Why the `|| data.isValueMap()` here?

Reply:
For some reason the `PersistedData` for a JSON object returns `true` for `isArray()` - and not only for actually JSON arrays as I originally thought. 

A JSON object is also a "value map", so we use this here to identify whether we are looking at the expected format or not.

Reply:
:grimacing: is that specific to JSON? is it a bug in PersistedData? 

This should be a straightforward precondition check, but I think the conditional is non-obvious, and that points to something that needs fixing elsewhere (in a separate issue).

Reply:
tracked in #5069